scripts/tests/chiptest/test_definition.py

```diff
@@ -142,30 +148,43 @@ def __startServer(self):
                     self.kvsPathSet.add(value)
         return self.runner.RunSubprocess(app_cmd, name='APP ', wait=False)
 
-    def __waitFor(self, waitForString, server_process, outpipe, timeoutInSeconds=10):
-        logging.debug('Waiting for %s' % waitForString)
+    def __waitFor(self, patterns: Iterable[str], process=None, outpipe: LogPipe | None = None, timeoutInSeconds: int = 10):
```

Comment:
Since `process` and `outpipe` are used from the `self` in all use cases (if I'm not mistaken) these method's args are not required at all. After all the `lastLogIndex` is kept in the self, so providing external process instance seems not right.